src/core/linalg/src/dense/4C_linalg_utils_densematrix_eigen.cpp

```diff
@@ -275,21 +270,40 @@ double Core::LinAlg::generalized_eigen(
    std::cout << std::setprecision(16) << "Q 2 " << Q_2 << std::endl;
    std::cout << std::setprecision(16) << "Z 2 " << Z_2 << std::endl;*/
 
-  double maxlambda = 0.;
+  std::vector<std::complex<double>> eigenvalues;
+  eigenvalues.reserve(N);
+  int num_inf = 0;
+
   for (int i = 0; i < N; ++i)
   {
-    if (BETA[i] > 1e-13)
+    if (std::abs(BETA[i]) < 1e-13)
     {
-      // Eigenvalues:
-      // std::cout << "lambda " << i << ":  " <<  ALPHAR[i]/BETA[i] << std::endl;
-      maxlambda = std::max(ALPHAR[i] / BETA[i], maxlambda);
+      ++num_inf;
+      continue;  // skip infinite eigenvalue
     }
     if (ALPHAI[i] > 1e-12)
     {
-      std::cout << " Warning: you have an imaginary EW " << ALPHAI[i] << std::endl;
+      std::cout << " Warning: you have an imaginary EW " << ALPHAI[i] << '\n';
     }
+    eigenvalues.emplace_back(ALPHAR[i] / BETA[i], ALPHAI[i] / BETA[i]);
+  }
+  if (num_inf > 0)
+    std::cout << "Generalized eigenvalue computation: " << num_inf
+              << " infinite eigenvalue(s) discarded." << '\n';
```

Comment:
Printing warnings in such low-level algorithms is not great since it limits usability. Can you return the data in a sorted way and remove the warning? Maybe you can have the real, imaginary, "infinite" etc. eigenvalues in separate vectors?